internal/proxy/impl.go

```diff
@@ -3959,3 +3959,184 @@ func (node *Proxy) GetImportState(ctx context.Context, req *milvuspb.GetImportSt
+	// validate params
+	username := req.Username
+	if err := ValidateUsername(username); err != nil {
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_IllegalArgument,
+			Reason:    err.Error(),
+		}, nil
+	}
+	rawPassword, err := crypto.Base64Decode(req.Password)
+	if err != nil {
+		log.Error("decode password fail", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_CreateCredentialFailure,
+			Reason:    "decode password fail key:" + req.Username,
+		}, nil
+	}
+	if err = ValidatePassword(rawPassword); err != nil {
+		log.Error("illegal password", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_IllegalArgument,
+			Reason:    err.Error(),
+		}, nil
+	}
+	encryptedPassword, err := crypto.PasswordEncrypt(rawPassword)
+	if err != nil {
+		log.Error("encrypt password fail", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_CreateCredentialFailure,
+			Reason:    "encrypt password fail key:" + req.Username,
+		}, nil
+	}
+	credInfo := &internalpb.CredentialInfo{
+		Username:          req.Username,
+		EncryptedPassword: encryptedPassword,
+	}
+	result, err := node.rootCoord.CreateCredential(ctx, credInfo)
+	if err != nil { // for error like conntext timeout etc.
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_UnexpectedError,
+			Reason:    err.Error(),
+		}, nil
+	}
+	return result, err
+}
+
+func (node *Proxy) UpdateCredential(ctx context.Context, req *milvuspb.CreateCredentialRequest) (*commonpb.Status, error) {
+	log.Debug("UpdateCredential", zap.String("role", typeutil.RootCoordRole), zap.String("username", req.Username))
+	// validate params
+	rawPassword, err := crypto.Base64Decode(req.Password)
+	if err != nil {
+		log.Error("decode password fail", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_UpdateCredentialFailure,
+			Reason:    "decode password fail when updating:" + req.Username,
+		}, nil
+	}
+	if err = ValidatePassword(rawPassword); err != nil {
+		log.Error("illegal password", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_IllegalArgument,
+			Reason:    err.Error(),
+		}, nil
+	}
+	encryptedPassword, err := crypto.PasswordEncrypt(rawPassword)
+	if err != nil {
+		log.Error("encrypt password fail", zap.String("username", req.Username), zap.Error(err))
+		return &commonpb.Status{
+			ErrorCode: commonpb.ErrorCode_UpdateCredentialFailure,
+			Reason:    "encrypt password fail when updating:" + req.Username,
+		}, nil
+	}
+	credInfo := &internalpb.CredentialInfo{
+		Username:          req.Username,
+		EncryptedPassword: encryptedPassword,
+	}
+	result, err := node.rootCoord.UpdateCredential(ctx, credInfo)
+	if err != nil { // for error like conntext timeout etc.
```

Comment:
Need to log error message when error out, for both proxy and rootcoord

Reply:
ok